File: SmartFood/Models/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartFood.Models
{
    public class Payment
    {
        public int Id { get; set; }
        [Column("order_id")]
        public int OrderId { get; set; }
        public string? Method { get; set; } 
        public decimal Amount { get; set; }
        public string? Status { get; set; } 
        [Column("transaction_id")]
        public string? TransactionId { get; set; }
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.Now;

        // Navigation
        public Order? Order { get; set; }
    }
}

File: smartfood_app/lib/models/payment.dart
class Payment {
  final int id;
  final int orderId;
  final String? method; // COD, VNPAY
  final double amount;
  final String? status; // pending, success
  final String? transactionId;
  final DateTime createdAt;

  Payment({
    required this.id,
    required this.orderId,
    this.method,
    required this.amount,
    this.status,
    this.transactionId,
    required this.createdAt,
  });

  // JSON -> Object
  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      orderId: json['order_id'],
      method: json['method'],
      amount: (json['amount'] as num).toDouble(),
      status: json['status'],
      transactionId: json['transaction_id'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  // Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'order_id': orderId,
      'method': method,
      'amount': amount,
      'status': status,
      'transaction_id': transactionId,
      'created_at': createdAt.toIso8601String(),
    };
  }
}